demo/repo/Jenkinsfile

```diff
@@ -22,6 +22,7 @@ stage('QA') {
 milestone 1
 stage('Staging') {
     lock(resource: 'staging-server', inversePrecedence: true) {
+        milestone 2
```

Comment:
@amuniz I found a problem in the following scenario:

* start build 1, wait for it to pause in `input`
* start build 2, wait for it to pause in `lock`
* start build 3, wait for it to pause in `lock`
* approve build 1, wait for it to finish
* approve build 3, wait for it to finish

Build 2 ought to have been aborted cleanly, but shows as failed since it started trying to deploy into staging, and was aborted in the middle of a step:

```
[Pipeline] milestone
Trying to pass milestone 1
[Pipeline] stage
[Pipeline] { (Staging)
[Pipeline] lock
Trying to acquire lock on [staging-server]
[staging-server] is locked, waiting...
Lock acquired on [staging-server]
[Pipeline] {
[Pipeline] node
Running on mock-agent-2 in /var/jenkins_home/mock-agents/mock-agent-2/workspace/cd/master
[Pipeline] {
[Pipeline] unstash
Superseded by cd/master#3
[Pipeline] }
[Pipeline] // node
[Pipeline] }
Lock released on resource [staging-server]
[Pipeline] // lock
[Pipeline] }
[Pipeline] // stage
[Pipeline] End of Pipeline
java.lang.InterruptedException
	at java.lang.Object.wait(Native Method)
	at hudson.remoting.Request.call(Request.java:147)
	at hudson.remoting.Channel.call(Channel.java:780)
	at hudson.FilePath.act(FilePath.java:1007)
	at hudson.FilePath.act(FilePath.java:996)
	at hudson.FilePath.untarFrom(FilePath.java:728)
	at org.jenkinsci.plugins.workflow.flow.StashManager.unstash(StashManager.java:124)
	at org.jenkinsci.plugins.workflow.support.steps.stash.UnstashStep$Execution.run(UnstashStep.java:64)
	at org.jenkinsci.plugins.workflow.support.steps.stash.UnstashStep$Execution.run(UnstashStep.java:54)
	at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1$1.call(AbstractSynchronousNonBlockingStepExecution.java:52)
	at hudson.security.ACL.impersonate(ACL.java:213)
	at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1.run(AbstractSynchronousNonBlockingStepExecution.java:49)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Finished: NOT_BUILT
```

After this fix, as soon as build 1 proceeds into production, build 2 is canceled (since 3 is already waiting), which is what I think we want:

```
[Pipeline] milestone
Trying to pass milestone 1
[Pipeline] stage
[Pipeline] { (Staging)
[Pipeline] lock
Trying to acquire lock on [staging-server]
[staging-server] is locked, waiting...
Superseded by cd/master#3
[Pipeline] // lock
[Pipeline] }
[Pipeline] // stage
[Pipeline] End of Pipeline
Finished: NOT_BUILT
```